frontend/src/lib/k8s/api/v2/hooks.ts

```diff
@@ -179,225 +183,22 @@ const getWorkingEndpoint = async (endpoints: KubeObjectEndpoint[], cluster: stri
  *
  * @params endpoints - List of possible endpoints
  */
-const useEndpoints = (endpoints: KubeObjectEndpoint[], cluster: string) => {
+export const useEndpoints = (
+  endpoints: KubeObjectEndpoint[],
+  cluster?: string,
+  namespace?: string
+) => {
   const { data: endpoint } = useQuery({
-    enabled: endpoints.length > 1,
+    enabled: endpoints.length > 1 && cluster !== undefined,
     queryKey: ['endpoints', endpoints],
     queryFn: () =>
-      getWorkingEndpoint(endpoints, cluster)
+      getWorkingEndpoint(endpoints, cluster!, namespace)
```

Comment:
Can you please explain why the typechecker is ignored for `cluster!` ? Maybe we need to handle this?

Reply:
cluster value is checked inside `enabled` variable, but TS doesn't know about that